### PR TITLE
Update NPC Generator case study lesson to use Google Gemini

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
[Asana Task](https://app.asana.com/0/1205655776549324/1208845256655000/f)

Update the last lesson under AI APIs so the code and descriptions match the updated NPC Generator code using Google Gemini. 

The return values for the routes in the code examples does not perfectly match the starter & solution branches at present. I opened a couple PRs to update the route return values in the NPC generator to match what we taught through C22's Unit 2 and am using that syntax in this lesson.
- [starter PR](https://github.com/AdaGold/NPC-Generator/pull/3/files)
- [solution PR](https://github.com/AdaGold/NPC-Generator/pull/4/files)

### CODE REVIEW

File: working-with-openai/open-ai-api-in-projects.md [hunk 1]
```diff
@@ -188,103 +189,109 @@ Create and connect your database using the steps below
...
-| <div style="min-width:300px;">Part of Code</div> | Description |
-| ------------- | ----------- | 
-| `from openai import OpenAI` | Import the OpenAI library |
-| `client = OpenAI()` | This code creates an OpenAI environment with your secret key from the `.env` file. We will use `client` to make requests to the OpenAI endpoints. |
+   
+   | <div style="min-width:300px;">Part of Code</div> | Description |
+   | ------------- | ----------- | 
+   | `import google.generativeai as genai` | Import the `google-generativeai` package and give it an alias of `genai`. |
+   | `import os` | Import the `os` package so we can read our `GEMINI_API_KEY` from the `.env` file. |
+   | `genai.configure(api_key=...)` | This code creates a Gemini API environment with your secret key from the `.env` file. We will use `genai` to access the `gemini-1.5-flash` model and make requests to the `/gemini-1.5-flash:generateContent` endpoint. |
```
Comment (kelsey-steven-ada) [inline working-with-openai/open-ai-api-in-projects.md]:
```suggestion
   | `genai.configure(api_key=...)` | This code configures the Gemini API environment with your secret key from the `.env` file. We will use the `genai` import to access the `gemini-1.5-flash` model and make requests to the `/gemini-1.5-flash:generateContent` endpoint. |
```

File: working-with-openai/open-ai-api-in-projects.md [hunk 2]
```diff
@@ -1,26 +1,28 @@
...
 
 Take some time to explore the NPC Generator repo in its current state. For this project, we will be finishing an API for a role-playing style game. Our API creates NPCs (Non Playable Characters) that your player's character will interact with. 
 
-We currently have two models, "Character" and "Greeting". "Character" holds some basic information about a character and has a one to many relationship with "Greeting", which represents a stock phrase that particular character may say when interacting with the player. Each NPC can have multiple greetings but each greeting can only be connected to one character. We will use the OpenAI API to generate those greetings which we will then store in our database.
+We currently have two models, "Character" and "Greeting". "Character" holds some basic information about a character and has a one to many relationship with "Greeting", which represents a stock phrase that particular character may say when interacting with the player. Each NPC can have multiple greetings but each greeting can only be connected to one character. We will use the Gemini API to generate those greetings which we will then store in our database.
```
Comment (apradoada) [inline working-with-openai/open-ai-api-in-projects.md]:
```suggestion
We currently have two models, "Character" and "Greeting". "Character" holds some basic information about a character and has a one to many relationship with "Greeting", which represents a stock phrase that a particular character may say when interacting with the player. Each NPC can have multiple greetings but each greeting can only be connected to one character. We will use the Gemini API to generate those greetings which we will then store in our database.
```

File: working-with-openai/open-ai-api-in-projects.md [hunk 3]
```diff
@@ -339,43 +344,32 @@ def get_greetings(char_id):
...
     
-    return jsonify(response)
+    return response
 ```
 </details>
 </br>
 
 ## Summary
-In this lesson, we took a partially built API and expanded it to include AI generated content. This is just the tip of the iceberg when it comes to integrating the OpenAI API into your projects, so we encourage you as always to follow your curiosity and keep looking for new and exciting ways to bring AI into your projects!
+In this lesson, we took a partially built API and expanded it to include AI generated content. This is just the tip of the iceberg when it comes to integrating the AI APIs into your code, so we encourage you as always to follow your curiosity and keep looking for new and exciting ways to bring AI into your projects!
```
Comment (apradoada) [inline working-with-openai/open-ai-api-in-projects.md:354]: Not sure if this is necessary, but would it be helpful to include a callout along the lines below?

```suggestion
### !callout-info

## Other AI APIs and Python
As mentioned before, Gemini is just one of many AI APIs you can integrate into your Python projects. We encourage you to explore others as your time and resources allow. While they won't look exactly the same, the steps for integration should be similar. There will likely be a Python package you can install with functions that will allow you to designate a model, attach your API key, make a request and more! Make sure to follow the documentation for whichever API you end up using and try things out!

### !end-callout  
```

**Review by apradoada — COMMENTED** [review]
Just a couple grammar suggestions. I also included a possible callout at the end to point out that not every API integration will look like this, but the steps will be pretty similar. Feel free to ignore it if you think it's not needed!

File: working-with-openai/open-ai-api-in-projects.md (hunk 3 above)
Comment (kelsey-steven-ada) [reply working-with-openai/open-ai-api-in-projects.md:354]: I think this is great, thank you for adding it!

File: working-with-openai/open-ai-api-in-projects.md [hunk 4]
```diff
@@ -339,43 +344,37 @@ def get_greetings(char_id):
...
+### !callout-info
```
Comment (kelsey-steven-ada) [inline working-with-openai/open-ai-api-in-projects.md:355]:
```suggestion

### !callout-info
```

File: working-with-openai/open-ai-api-in-projects.md [hunk 5]
```diff
@@ -339,43 +344,38 @@ def get_greetings(char_id):
...
+In this lesson, we took a partially built API and expanded it to include AI generated content. This is just the tip of the iceberg when it comes to integrating the AI APIs into your code, so we encourage you as always to follow your curiosity and keep looking for new and exciting ways to bring AI into your projects!
+
+### !callout-info
 
+## Other AI APIs and Python
+As mentioned before, Gemini is just one of many AI APIs you can integrate into your Python projects. We encourage you to explore others as your time and resources allow. While they won't look exactly the same, the steps for integration should be similar. There will likely be a Python package you can install with functions that will allow you to designate a model, attach your API key, make a request and more! Make sure to follow the documentation for whichever API you end up using and try things out!
+
+### !end-callout  
 ## Check For Understanding
```
Comment (kelsey-steven-ada) [inline working-with-openai/open-ai-api-in-projects.md]:
```suggestion
### !end-callout  

## Check For Understanding
```

**Review by apradoada — APPROVED** [review]

**Pull request merged**